### DISCUSSION TITLE
如何在描述域参数的opt部分使用内置变量？

### DISCUSSION THREAD

**Discussion by FZerolight** [xmake-io/xmake#3971] [Q&A] [answered]
现在有一个需求是，在当前工程目录下有一个"3rdparty"文件夹，里面存放了一些cmake库，例如libSDL2。在使用add_requires加载cmake库时为了能正确搜索到依赖，想使用`{configs = {envs = {CMAKE_PREFIX_PATH = "$(projectdir)/3rdparty/SDL2"}}}`来显式指定库路径，但似乎`$(projectdir)`的在这部分并不会被替换，具体的内容如下：

```
add_requires("cmake::SDL2", {configs = {envs = {CMAKE_PREFIX_PATH = "$(projectdir)/3rdparty/SDL2"}}})
```

想问一下在设计上这种需求该如何正确的实现？

**Comment by waruqi** [comment]
先得确认 3rdparty 里面放的是 二进制库，还是源码库。。处理方式不同的。。

**Reply by FZerolight** [reply]
二进制库，现在写成`CMAKE_PREFIX_PATH = "D:/foo/bar/3rdparty/SDL2"`是可以正常工作的

**Reply by waruqi** [reply]
二进制库，直接 add_linkdirs add_links add_includedirs 不就好了么。。还去走 cmake 绕下干嘛

**Reply by star-hengxing** [reply]
二进制包用 package，然后走 on fetch 就行

**Reply by FZerolight** [reply, marked as answer]
> 二进制库，直接 add_linkdirs add_links add_includedirs 不就好了么。。还去走 cmake 绕下干嘛

好像非常合理……主要是想捡现成的cmake配置，包括一些宏和依赖项。现在因为出现了新需求所以我干脆直接重新写package了
